tests/xdc/stream_based_replication_test.go

```diff
@@ -0,0 +1,379 @@
+		namespaceName string
+		namespaceID   string
+		serializer    serialization.Serializer
+		generator     test.Generator
+	}
+)
+
+func TestStreamBasedReplicationTestSuite(t *testing.T) {
+	suite.Run(t, new(streamBasedReplicationTestSuite))
+}
+
+func (s *streamBasedReplicationTestSuite) SetupSuite() {
+	s.controller = gomock.NewController(s.T())
+	s.dynamicConfigOverrides = map[dynamicconfig.Key]interface{}{
+		dynamicconfig.EnableReplicationStream:       true,
+		dynamicconfig.EnableEagerNamespaceRefresher: true,
+		dynamicconfig.EnableReplicationTaskBatching: true,
+	}
+	s.logger = log.NewNoopLogger()
+	s.serializer = serialization.NewSerializer()
+	s.setupSuite(
+		[]string{
+			"active",
+			"standby",
+		},
+		tests.WithFxOptionsForService(primitives.AllServices,
+			fx.Decorate(
+				func() config.DCRedirectionPolicy {
+					return config.DCRedirectionPolicy{Policy: "noop"}
+				},
+			),
+		),
+	)
+	ctx := context.Background()
+	s.namespaceName = "replication-test"
+	s.cluster1.GetFrontendClient().RegisterNamespace(ctx, &workflowservice.RegisterNamespaceRequest{
+		Namespace: s.namespaceName,
+		Clusters:  s.clusterReplicationConfig(),
+		// The first cluster is the active cluster.
+		ActiveClusterName: s.clusterNames[0],
+		// Needed so that the namespace is replicated.
+		IsGlobalNamespace: true,
+		// This is a required parameter.
+		WorkflowExecutionRetentionPeriod: durationpb.New(time.Hour * 24),
+	})
+	err := s.waitUntilNamespaceReplicated(ctx, s.namespaceName)
+	s.Require().NoError(err)
+
+	nsRes, _ := s.cluster1.GetFrontendClient().DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: s.namespaceName,
+	})
+
+	s.namespaceID = nsRes.NamespaceInfo.GetId()
+	s.generator = test.InitializeHistoryEventGenerator("namespace", "ns-id", 1)
+}
+
+func (s *streamBasedReplicationTestSuite) TearDownSuite() {
+	if s.generator != nil {
+		s.generator.Reset()
+	}
+	s.controller.Finish()
+	s.tearDownSuite()
+}
+
+func (s *streamBasedReplicationTestSuite) SetupTest() {
+	s.setupTest()
+	// Register a namespace.
+
+}
+
+func (s *streamBasedReplicationTestSuite) TestReplicateHistoryEvents_ForceReplicationScenario() {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, testTimeout)
+	defer cancel()
+
+	// let's import some events into cluster 1
+	historyClient1 := s.cluster1.GetHistoryClient()
+	executions := s.importTestEvents(historyClient1, namespace.Name(s.namespaceName), namespace.ID(s.namespaceID), []int64{3, 13, 2, 202, 302, 402, 602, 502, 802, 1002, 902, 702, 1102})
+
+	// let's trigger replication by calling GenerateLastHistoryReplicationTasks. This is also used by force replication logic
+	for _, execution := range executions {
+		_, err := historyClient1.GenerateLastHistoryReplicationTasks(ctx, &historyservice.GenerateLastHistoryReplicationTasksRequest{
+			NamespaceId: s.namespaceID,
+			Execution:   execution,
+		})
+		s.NoError(err)
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
this can delay test running. do we expect to run this test for every check-in?

Reply:
Yeah, this is running for every check-in. I will leave as it for now. I will try to think if there is a way to improve it.